cmd/autoconfigbrancher/main.go

```diff
@@ -265,3 +259,39 @@ func main() {
 		logrus.WithError(err).Fatal("PR creation failed.")
 	}
 }
+
+func runSteps(steps []step, author string, stdout, stderr io.Writer) (needsPushing bool, err error) {
+	startCommitOut, err := exec.Command("git", "rev-parse", "HEAD").CombinedOutput()
+	if err != nil {
+		return false, fmt.Errorf("failed to execute `git rev-parse HEAD`: %w\noutput:%s\n", err, string(startCommitOut))
+	}
+	startCommitSHA := strings.TrimSpace(string(startCommitOut))
+
+	var commitsCounter int
```

Comment:
All this tracking code seems redundant now since the inexpensive `git` call at the end will give us the same information.

Reply:
Execing out is all but inexpensive :P

Reply:
"Inexpensive" with respect to the operation itself, which is essentially a hash comparison.

I don't see the need to have an extra return value for the function and a counter (which doesn't need to be a counter to begin with) if the result will be the same as letting `git` handle it with negligible cost, all in a single code path.

Reply:
Because being able to differentiate those two cases might be interesting, ideally the second case would never happen.